pkg/controllers/jobset_controller.go

```diff
@@ -218,6 +224,69 @@ func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet)
 	}
 	return &ownedJobs, nil
 }
+func (r *JobSetReconciler) calculateAndUpdateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) error {
+	status, err := r.calculateReplicatedJobStatuses(ctx, js, jobs)
+	if err != nil {
+		return err
+	}
+	// Check if status ReplicatedJobsStatus has changed
+	if apiequality.Semantic.DeepEqual(js.Status.ReplicatedJobsStatus, status) {
+		return nil
+	}
+	js.Status.ReplicatedJobsStatus = status
+	return r.Status().Update(ctx, js)
+}
+
+func (r *JobSetReconciler) calculateReplicatedJobStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) ([]jobset.ReplicatedJobStatus, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Prepare replicatedJobsReady for optimal iteration
+	replicatedJobsReady := map[string]int32{}
+	for _, replicatedJob := range js.Spec.ReplicatedJobs {
+		replicatedJobsReady[replicatedJob.Name] = 0
+	}
+
+	// Calculate jobsReady for each Replicated Job
+	for _, job := range append(jobs.active, append(jobs.successful, jobs.failed...)...) {
```

Comment:
successful and failed jobs should not be counted as "ready", we should only look at active.

but we also need to iterate over successful in a second loop to calculate the number of Succeeded jobs